File: Tuesdays-content/Question1.js
function solution(R, V) {
    
    let balanceA = 0;
    let balanceB = 0;
    
    let minBalanceA = 0;
    let minBalanceB = 0;
    
    for (let i = 0; i < R.length; i++) {
        if (R[i] === 'A') {
            balanceB -= V[i];
            minBalanceB = Math.min(minBalanceB, balanceB);
        } else if (R[i] === 'B') {
            
            balanceA -= V[i];
    
            minBalanceA = Math.min(minBalanceA, balanceA);
        }
    }
    
   
    return [Math.max(0, -minBalanceA), Math.max(0, -minBalanceB)];
}

// Example usage
const R1 = "BAABA";
const V1 = [2, 4, 1, 1, 2];
console.log(solution(R1, V1));  // Expected output: [2, 4]

const R2 = "ABAB";
const V2 = [10, 5, 10, 15];
console.log(solution(R2, V2));  // Expected output: [0, 15]

const R3 = "B";
const V3 = [100];
console.log(solution(R3, V3));  // Expected output: [100, 0]

File: Tuesdays-content/Question1.py
def solution(R, V, N):
    current_balance_A = 0
    current_balance_B = 0
    min_balance_A = 0
    min_balance_B = 0

    for i in range(N):
        if R[i] == 'A':
            current_balance_A += V[i]
            current_balance_B -= V[i]
        else:  # R[i] == 'B'
            current_balance_B += V[i]
            current_balance_A -= V[i]

        if current_balance_A < min_balance_A:
            min_balance_A = current_balance_A
        if current_balance_B < min_balance_B:
            min_balance_B = current_balance_B

    initial_balance_A = max(0, -min_balance_A)
    initial_balance_B = max(0, -min_balance_B)

    return [initial_balance_A, initial_balance_B]


R = "ABAB"
V = [10, 5, 10, 15]
N = len(R)
print(solution(R, V, N))  # Output should be [0, 15]